tests/TestCase/Controller/ControllerTest.php

```diff
@@ -1097,24 +1126,50 @@ public function testBeforeRenderViewVariables()
      */
     public function testDeprecatedViewProperty($property, $getter, $setter, $value)
     {
-        $controller = new AnotherTestController();
-        $message = false;
-
-        set_error_handler(function ($errno, $errstr) use (&$message) {
-            $message = ($errno === E_USER_DEPRECATED ? $errstr : false);
-        });
-
-        try {
+            $controller = new AnotherTestController();
+            error_reporting(E_ALL ^ E_USER_DEPRECATED);
             $controller->$property = $value;
-            $this->assertSame(sprintf('Controller::$%s is deprecated. Use $this->viewBuilder()->%s() instead.', $property, $setter), $message);
-
             $this->assertSame($value, $controller->$property);
-            $this->assertSame(sprintf('Controller::$%s is deprecated. Use $this->viewBuilder()->%s() instead.', $property, $getter), $message);
-
             $this->assertSame($value, $controller->viewBuilder()->{$getter}());
-        } finally {
-            restore_error_handler();
-        }
+    }
+
+    /**
+     * Tests deprecated view propertiyes message
+     *
+     * @param $property Deprecated property name
+     * @param $getter Getter name
+     * @param $setter Setter name
+     * @param mixed $value Value to be set
+     * @return void
+     * @expectedException PHPUnit\Framework\Error\Deprecated
+     * @expectedExceptionMessageRegExp /^Controller::\$\w+ is deprecated(.*)/
+     * @dataProvider deprecatedViewPropertyProvider
+     */
+    public function testDeprecatedViewPropertySetterMessage($property, $getter, $setter, $value)
```

Comment:
You don't use getter ever. Perhaps it should be removed?

Reply:
Yes,but why this? because I use exist/and same data provider for 3 test

Reply:
by using same data(provider) we can ensure all test is ok when we want add `viewVarsTrait::viewClass` to this data provider future

Reply:
I see again 
there are 3 test with bellow usage
```php
1) ($property, $getter, null, $value)
2) ($property, null, null, $value)
3) ($property, null, $setter, $value)
```
we can remove some of args just by changing order of args

Reply:
Probably not worth the effort.